modules/core/24-host/parse.go

```diff
@@ -27,7 +27,7 @@ func ParseIdentifier(identifier, prefix string) (uint64, error) {
 
 	sequence, err := strconv.ParseUint(splitStr[1], 10, 64)
 	if err != nil {
-		return 0, errorsmod.Wrap(err, "failed to parse identifier sequence")
+		return 0, errorsmod.Wrap(ErrInvalidID, err.Error())
```

Comment:
In these scenario (where there is an actual error), we typically don't wrap like this, we wrap the way this was already doing: `errorsmod.Wrap(err, "failed to parse identifier sequence")`

Reply:
-Consistency: Other errors in the same function are using ErrInvalidID instead of wrapping directly from err. This ensures that the error message is always consistent about the error type ErrInvalidID.

-Error structure: ErrInvalidID is used to represent a specific identifier format error. If wrapped with another external err (errorsmod.Wrap), it may not ensure consistency about the error type.

I suggest the following:
`return 0, errorsmod.Wrapf(ErrInvalidID, "failed to parse sequence: %s", err.Error())`

Are you ok?

Reply:
For this case either is fine imo, but generally prefer to wrap err returned as @gjermundgaraba suggested

Reply:
Thanks u

> For this case either is fine imo, but generally prefer to wrap err returned as @gjermundgaraba suggested